lib/srv/db/cloud/iam.go

```diff
@@ -168,10 +169,15 @@ func (c *IAM) UpdateIAMStatus(database types.Database) error {
 
 		iamPolicyStatus, ok := c.iamPolicyStatus.Load(database.GetName())
 		if !ok {
-			iamPolicyStatus = false
+			// If there was no key found it was a result of un-registering database
+			// (and policy) as a result of deletion or failing to re-register from
+			// updating database.
+			awsStatus.IAMPolicyStatus = types.IAMPolicyStatus_IAM_POLICY_STATUS_UNSPECIFIED
```

Comment:
I'm not sure I fully understand this. In which case do we need to set it to "unspecified" and why? Do we even need to bother updating it here if it wasn't found?

Reply:
tbh i don't think it's necessary. we don't really care about the `policy status` near tear down, i also think it's unlikely to reach here after tearing down.